src/components/cart-checkout-card/cart-checkout-card.jsx

```diff
@@ -0,0 +1,70 @@
+import "../cartCard/cart-card.css";
+import { useCartContext } from "../../contexts/Cart-context/cart-context";
+import { React, Fragment, useState, useEffect } from "react";
+import { calculateCartItems } from "../../utils/cart-details";
+const CartCheckout = () => {
+  const { cartState } = useCartContext();
+  const [cartDetails, setCartDetails] = useState(0);
+
+  useEffect(() => {
+    const getDetails = cartState.reduce(
+      (acum, cur) => {
+        acum.CartQty += cur.qty;
+        acum.totalMrp += cur.price * cur.qty;
+        const [discountedPrice, sumToMinus] = calculateCartItems(cur.price, 40);
+        acum.discountedPrice += sumToMinus * cur.qty;
+        acum.subTotal += discountedPrice * cur.qty;
+        return acum;
+      },
+      { CartQty: 0, totalMrp: 0, discountedPrice: 0, subTotal: 0 }
```

Comment:
You can directly call this function no need to use useEffect and state

Reply:
Sure, will do that change